pkg/controller/migcluster/validation.go

```diff
@@ -346,3 +357,65 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	clusterClient, err := cluster.GetClient(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	clusterOperatorVersion, err := cluster.GetOperatorVersion(clusterClient)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	if clusterOperatorVersion == "" {
+
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     ClusterOperatorVersionNotFound,
+			Status:   True,
+			Reason:   VersionNotFound,
+			Category: Warn,
+			Message:  fmt.Sprintf(`MTC operator version lookup failed. Key %s not found in configmap %s/%s. Ensure that all clusters are running the same MTC Operator version.`, migapi.OperatorVersionKey, migapi.VeleroNamespace, migapi.ClusterConfigMapName),
+		})
+
+		return liberr.Wrap(err)
+	}
+
+	hostOperatorVersion, err := cluster.GetOperatorVersion(r)
+	if err != nil {
+		return liberr.Wrap(err)
+	}
+	if hostOperatorVersion == "" {
+
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     ClusterOperatorVersionNotFound,
+			Status:   True,
+			Reason:   VersionNotFound,
+			Category: Warn,
+			Message:  fmt.Sprintf(`MTC operator version lookup failed. Key %s not found in configmap %s/%s. Ensure that all clusters are running the same MTC Operator version.`, migapi.OperatorVersionKey, migapi.VeleroNamespace, migapi.ClusterConfigMapName),
+		})
+
+		return liberr.Wrap(err)
+	}
+
+	operatorVersionMatchesHost := clusterOperatorVersion == hostOperatorVersion
+
+	if !operatorVersionMatchesHost {
+		cluster.Status.SetCondition(migapi.Condition{
+			Type:     OperatorVersionMismatch,
+			Status:   True,
+			Reason:   VersionCheckFailed,
+			Category: Warn,
```

Comment:
@djwhatle Ignore if you have already thought about this. It feels like this should be a Critical condition instead of Warn

Reply:
We decided `Warning` was appropriate here for now. I think when we get a bit of mileage on this system and know it won't malfunction I'd be more willing to make it a hard blocker.